### DISCUSSION TITLE
Math from scratch in Haskell: addition | Loominate

### DISCUSSION THREAD

**Discussion by giscus** [michaelgwelch/loominate.net#8] [Blog Comments]
# Math from scratch in Haskell: addition | Loominate

Last time I developed a representation of Natural numbers following Eric Lippert’s lead. In this post I’ll continue to follow him and implement addition.

https://loominate.net/2013/10/11/math-from-scratch-in-haskell-addition/

**Comment by michaelgwelch** [comment]
Michael, check your work or explain it better. This is future you, who is going to recreate and test this because it’s not obvious the addition function is correct.

**Reply by michaelgwelch** [reply]
There needs to be clarity: in the `Nat` constructor is the new bit going in the least significant bit or most significant bit.
